patterns/1-initial/explicit-shared-ownership.md

```diff
@@ -0,0 +1,51 @@
+# Title
+
+Explicit shared ownership
+
+# Patlet
+
+In a software system there is a component that several teams depend on. Initially developed by a handful of people it has long outgrown it's orignal use-case. Original authors are no longer capable of taking full ownership of that component. Development happens organically. Often there is confusion as to who should make changes, who needs to approve changes. Making ownership explicit, sharing that ownership across dependent teams, making expected behaviour explicit helps remove ambiguity. Writing those learnings down means that it can be changed according to the same process that is used to change the software.
+
+# Problem
+
+An organisation is already using InnerSource best practices in several teams. The architecture of the software offered has grown organically. Talking about code ownership and accountability teams notice that there is a component that is in a worrying state: Developed by an original team of authors it has grown it's userbase way beyond what the original team can handle. From time to time others step up to help out, however there is no formal process established. As a result conflicts arise around who should do the work, who should decide on project direction. Simply forking the component would lead to a lot of duplication and wasted effort. As a result teams are looking for a less intrusive solution to the issue.
```

Comment:
> An organisation is already using InnerSource best practices in several teams.

I did actually observe this mostly even without InnerSource being known or best practices applied. The effects described are in force nevertheless.

There is an extension to this here where a project is actively abandoned or it's existence is denied in fear of being hit with the maintenance while it still is a crucial part of a given system. The way to fix this involves the same route given here.

Reply:
I agree that this occurs also outside of the InnerSource context. I believe though that it is important to not start rolling out InnerSource by addressing this type of problem: The natural reflex in traditional corporations seems to be more a top down approach to solving this issue. The approach described in this pattern though is much more bottom up. I believe it needs a bit of experience to make this pattern work in practice.

Reply:
@MaineC based on your last comment it would make sense to add this to the Context section, as this sounds like a requirement to successfully apply this pattern:

* The organisation is already using InnerSource best practices in several teams. (i.e. the ideas of InnerSource are not new for the organisation)

Reply:
Hmm - wouldn't we have to add this to pretty much all of the patterns (except maybe the "start with an experiment" and similar ones)?

Reply:
Yeah, that is also true. So let's just leave things as they are here.